lib/resty/upstream/healthcheck.lua

```diff
@@ -651,6 +651,78 @@ local function gen_peers_status_info(peers, bits, idx)
     return idx
 end
 
+local function gen_peers_prometheus_status_info(peers, bits, idx, u, role)
+    local npeers = #peers
+    for i = 1, npeers do
+        idx = idx + 1
+        local peer = peers[i]
+        if peer.down then
+            bits[idx] = string.format("nginx_upstream_status_info{name=\"%s\",endpoint=\"%s\", status=\"DOWN\", role=\"%s\"} 1\n", u, peer.name, role)
+        else
+            bits[idx] = string.format("nginx_upstream_status_info{name=\"%s\",endpoint=\"%s\", status=\"UP\", role=\"%s\"} 1\n", u, peer.name, role)
+        end
+    end
+    return idx
+end
+
+function _M.prometheus_status_page()
+    -- generate an prometheus metrics
+    -- # HELP nginx_upstream_status The running staus of nginx upstream
+    -- # TYPE nginx_upstream_status gauge
+    -- nginx_upstream_status{name="",endpoint="",status="", role=""} num
+
+    local us, err = get_upstreams()
+    if not us then
+        return "failed to get upstream names: " .. err
```

Comment:
I think it would be better to return `nil, err` in prometheus page, so the caller can check the result and log the error.

Reply:
Tks for you suggest. I have modified it in the latest commit.